### PR TITLE
Don't override Span/Transaction status on Finish(...) if status was not provided explicitly

### LINKED ISSUE #919: transaction.Finish() overrides the transaction status set via the public setter with the default parameter (Ok).
repro:

var tran = startTransaction();
tran.Status = SpanStatus.Error;
tran.Finish();

That's due to:
public void Finish(SpanStatus status = SpanStatus.Ok)

Suggestion: Have a proper parameterless method.
Also, don't set Ok if nothing was set. Sentry has a status: 'unknown' when nothing is set. Verify the spec/other SDKs if they default to OK or just send it without status.

### PR DESCRIPTION
**Pull request by Tyrrrz**
Fixes #919

### CODE REVIEW

File: test/Sentry.Tests/HubTests.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public void PushAndLockScope_DoesNotAffectOuterScope()
             hub.ConfigureScope(s => Assert.False(s.Locked));
         }
 
-        [Fact]
+        [Fact(Skip = "Flaky")]
```
Comment (bruno-garcia) [inline test/Sentry.Tests/HubTests.cs:74]: Best to raise issues to address these later, if there are no issues yet

**Review by bruno-garcia — APPROVED** [review]
We need a big note on the release notes once we ship since this will affect transactions in Sentry. Currently users would be getting OK for simply calling `finish` now it'll be `unknown`.

A Migration note on the docs will be needed to: From version x to y: Before: Finish() -> Now you need to explicitly call Finish(Ok)

File: test/Sentry.Tests/HubTests.cs (hunk 1 above)
Comment (Tyrrrz) [reply test/Sentry.Tests/HubTests.cs:74]: There are issues about specific tests, maybe one of them can be generalized.

**Comment by bruno-garcia** [comment]
can be merged?

**Comment by bruno-garcia** [comment]
develop discussion about this: https://github.com/getsentry/develop/issues/298

**Comment by Tyrrrz** [comment]
@bruno-garcia yes

**Pull request merged**